player/lua/select.lua

```diff
@@ -326,6 +298,54 @@ mp.add_key_binding(nil, "select-subtitle-line", function ()
             mp.commandv("seek", sub_times[index] + delay, "absolute")
         end,
     })
+end
+
+mp.add_key_binding(nil, "select-subtitle-line", function ()
+    local sub = mp.get_property_native("current-tracks/sub")
+
+    if sub == nil then
+        show_warning("No subtitle is loaded.")
+        return
+    end
+
+    if sub.external and sub["external-filename"]:find("^edl://") then
+        sub["external-filename"] = sub["external-filename"]:match('https?://.*')
+                                   or sub["external-filename"]
+    end
+
+    local screenx, screeny = mp.get_osd_size()
+    local osd_align_x = mp.get_property("osd-align-x")
+    if not sub.external or sub["external-filename"]:find("://") then
+        mp.osd_message("Subtitle export in progress.", 3)
+        mp.set_osd_ass(screenx, screeny, string.format("{\\an%d}● ",
+            osd_align_x == "right" and 7 or 9))
+    end
+
+    mp.command_native_async({
+        name = "subprocess",
+        capture_stdout = true,
+        args = sub.external
+            and {"ffmpeg", "-loglevel", "error", "-i", sub["external-filename"],
+                 "-f", "lrc", "-map_metadata", "-1", "-fflags", "+bitexact", "-"}
+            or {"ffmpeg", "-loglevel", "error", "-i", mp.get_property("path"),
+                "-map", "s:" .. sub["id"] - 1, "-f", "lrc", "-map_metadata",
+                "-1", "-fflags", "+bitexact", "-"}
+    }, function (success, result)
+        if not sub.external or sub["external-filename"]:find("://") then
+            mp.osd_message("")
+            mp.set_osd_ass(screenx, screeny, "")
```

Comment:
Can be shortened `mp.set_osd_ass(0, 0, "")`, which allows retrieving `mp.get_osd_size()` only within the previous `if` block.

Reply:
Done.